[SIL] support undef cases in switch_value instructions

ezephir opened the pull request:
This change adds support for undefined cases in the switch_value
instruction. Fixes SR-210.

lattner commented:
Nadav, it doesn't look like manman has a github account, can you please talk to her?

nadavrot commented:
@lattner Will do. 

@ezephir Thanks for working on this. We'll have someone review your pull request as soon as possible.

ezephir commented:
Fixed. Thanks!

ezephir commented:
Updated again to run the output through 'sil-opt' twice instead of invoking the swift front-end.

mren2 commented:
Thanks for working on this. Looking at the patch now.

mren2 commented on lib/Parse/ParseSIL.cpp:
```diff
@@ -3138,32 +3138,36 @@ bool SILParser::parseSILInstruction(SILBasicBlock *BB) {
-          if (!FR) {
-            if (auto *CF = dyn_cast<ConvertFunctionInst>(CaseVal)) {
-              FR = dyn_cast<FunctionRefInst>(CF->getOperand());
+          // are function references or undef.
```
Can you please add a testing case for functionTy as well? Thanks,

mren2 commented on test/SIL/Parser/undef.sil:
```diff
@@ -0,0 +1,17 @@
+// CHECK-LABEL: sil @undef_in_switch_value_case : $@convention(thin) () -> ()
+sil @undef_in_switch_value_case : $@convention(thin) () -> () {
+bb0:
+	%0 = integer_literal $Builtin.Int1, 0
```
I think we prefer 2 whitespaces as indent.

ezephir commented:
Okay. So I've:
* converted from hard tabs to soft (2 spaces per)
* added checks for the function case
* rebased this change against master

The pull request is up-to-date with fixes for all the feedback given so far. Thanks!
